homeassistant/components/fronius/__init__.py

```diff
@@ -220,20 +230,12 @@ async def _get_inverter_infos(self) -> list[FroniusDeviceInfo]:
 
         try:
             _inverter_info = await self.fronius.inverter_info()
-        except FroniusError as err:
+        except FroniusError:
             if self.config_entry.state == ConfigEntryState.LOADED:
                 # During a re-scan we will attempt again as per schedule.
                 _LOGGER.debug("Re-scan failed for %s", self.host)
                 return inverter_infos
-
-            raise ConfigEntryNotReady(
-                translation_domain=DOMAIN,
-                translation_key="entry_cannot_connect",
-                translation_placeholders={
-                    "host": self.host,
-                    "fronius_error": str(err),
-                },
-            ) from err
+            raise  # handled in async_setup_entry when not LOADED
```

Comment:
This looks like an unneeded move, to be honest?

Reply:
this is the reason: https://github.com/home-assistant/core/pull/131770#issuecomment-2543496280

Reply:
I'll take a look to see if the check can be improved.